include/eve/module/core/regular/impl/simd/riscv/last_true.hpp

```diff
@@ -0,0 +1,43 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+*/
+//==================================================================================================
+#pragma once
+
+#include <eve/arch/top_bits.hpp>
+#include <eve/module/core/regular/any.hpp>
+#include <eve/module/core/regular/convert.hpp>
+
+namespace eve::detail
+{
+
+template<arithmetic_scalar_value T, typename N, relative_conditional_expr C>
+EVE_FORCEINLINE std::optional<std::ptrdiff_t>
+last_true_(EVE_SUPPORTS(rvv_), C const& cond, top_bits<logical<wide<T, N>>> const& v0) noexcept
+requires rvv_abi<abi_t<T, N>>
+{
+  if constexpr( C::is_complete && !C::is_inverted ) return std::nullopt;
+  else
+  {
+    auto                m      = cond.mask(as<wide<T, N>> {});
+    logical<wide<T, N>> masked = __riscv_vmand(v0.storage, m, N::value);
+    return last_true(masked);
+  }
+}
+
+template<arithmetic_scalar_value T, typename N>
+EVE_FORCEINLINE std::optional<std::ptrdiff_t>
+                last_true_(EVE_SUPPORTS(rvv_), top_bits<logical<wide<T, N>>> const                &v0) noexcept
+requires rvv_abi<abi_t<T, N>>
+{
+  // TODO: optimize.
+  for( int i = N::value - 1; i > -1; --i )
+  {
+    if( v0.get(i) ) return i;
+  }
+  return {};
+}
```

Comment:
I am not wild about doing this tbh. Is there really no good solution in the isa? should be smth easy-ish. 

You can iota + mask + maximum for example.

Reply:
Actually, I suppose we can be faster, but I would prefer to make it in different MR, as this will require some investigation.

For example, I was thinking about casting it to several u64, and find it there. Unfortunately, I could not find in RVV ISA instruction that allows me to find position of  the last set bit.